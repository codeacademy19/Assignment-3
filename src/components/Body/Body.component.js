import React, { Component } from 'react';
import AllBooksOfAnAuthor from '../AllBooksOfAnAuthor/AllBooksOfAnAuthor.component';


class AllPost extends Component {
  render() {
    // const posts = this.props.posts;
    // console.log(this.props.blogs);
    return (

      <div className="allBooks">

        <AllBooksOfAnAuthor books={this.props.books['J K Rowling']} author="J.K.Rowling" />
        <AllBooksOfAnAuthor books={this.props.books['Sidney Sheldon']} author="Sidney Sheldon" />

      </div>
    );
  }
}
export default AllPost;
