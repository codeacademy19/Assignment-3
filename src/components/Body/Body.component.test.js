import React from 'react';
import renderer from 'react-test-renderer';
import Body from './Body.component';


describe('AllBooksOfAnAuthor', () => {
  it('should render without crashing', () => {
    const books = { 'J K Rowling': [{ Name: 'Harry Potter and the Sorcerers Stone (Harry Potter, #1)', liked: false, rating: 4.5 }, { Name: 'Harry Potter and the Sorcerers Stone (Harry Potter, #1)', liked: false, rating: 4.5 }], 'Sidney Sheldon': [{ Name: 'Harry Potter and the Sorcerers Stone (Harry Potter, #1)', liked: false, rating: 4.5 }, { Name: 'Harry Potter and the Sorcerers Stone (Harry Potter, #1)', liked: false, rating: 4.5 }] };
    const tree = renderer.create(<Body author="J.K.ROWLING" books={books} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
