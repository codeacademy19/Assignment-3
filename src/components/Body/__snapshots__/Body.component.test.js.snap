// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllBooksOfAnAuthor should render without crashing 1`] = `
<div
  className="allBooks"
>
  <fieldset
    className="allBooksOfAnAuthor"
  >
    <legend
      className="authorName"
    >
      J.K.Rowling
    </legend>
    <div
      className="allBooksMain"
    >
      <div
        className="BookPost"
      >
        <div
          className="main"
        >
          <div
            className="backgroundImage"
            style={
              Object {
                "backgroundImage": "url(\\"./Images/cover_image.jpeg\\")",
              }
            }
          />
          <div
            className="title"
          >
            <span>
              HARRY POTTER AND THE SORCERERS STONE (HARRY POTTER, #1)
            </span>
          </div>
        </div>
        <div
          className="reaction"
        >
          <span
            className="rating"
          >
            4.5
          </span>
          <img
            alt="heart"
            className="heart"
            src="./Icons/heart-black.svg"
          />
        </div>
      </div>
      <div
        className="BookPost"
      >
        <div
          className="main"
        >
          <div
            className="backgroundImage"
            style={
              Object {
                "backgroundImage": "url(\\"./Images/cover_image.jpeg\\")",
              }
            }
          />
          <div
            className="title"
          >
            <span>
              HARRY POTTER AND THE SORCERERS STONE (HARRY POTTER, #1)
            </span>
          </div>
        </div>
        <div
          className="reaction"
        >
          <span
            className="rating"
          >
            4.5
          </span>
          <img
            alt="heart"
            className="heart"
            src="./Icons/heart-black.svg"
          />
        </div>
      </div>
    </div>
  </fieldset>
  <fieldset
    className="allBooksOfAnAuthor"
  >
    <legend
      className="authorName"
    >
      Sidney Sheldon
    </legend>
    <div
      className="allBooksMain"
    >
      <div
        className="BookPost"
      >
        <div
          className="main"
        >
          <div
            className="backgroundImage"
            style={
              Object {
                "backgroundImage": "url(\\"./Images/cover_image.jpeg\\")",
              }
            }
          />
          <div
            className="title"
          >
            <span>
              HARRY POTTER AND THE SORCERERS STONE (HARRY POTTER, #1)
            </span>
          </div>
        </div>
        <div
          className="reaction"
        >
          <span
            className="rating"
          >
            4.5
          </span>
          <img
            alt="heart"
            className="heart"
            src="./Icons/heart-black.svg"
          />
        </div>
      </div>
      <div
        className="BookPost"
      >
        <div
          className="main"
        >
          <div
            className="backgroundImage"
            style={
              Object {
                "backgroundImage": "url(\\"./Images/cover_image.jpeg\\")",
              }
            }
          />
          <div
            className="title"
          >
            <span>
              HARRY POTTER AND THE SORCERERS STONE (HARRY POTTER, #1)
            </span>
          </div>
        </div>
        <div
          className="reaction"
        >
          <span
            className="rating"
          >
            4.5
          </span>
          <img
            alt="heart"
            className="heart"
            src="./Icons/heart-black.svg"
          />
        </div>
      </div>
    </div>
  </fieldset>
</div>
`;
